File: src/dataset/javascript/prime-factors_exercism-elenamountz/gemini/GeminiFlash_prime-factors_v1.js
export const primeFactors = (number) => {
  const factors = [];
  let divisor = 2;
  while (divisor * divisor <= number) {
    while (number % divisor === 0) {
      factors.push(divisor);
      number /= divisor;
    }
    divisor++;
  }
  if (number > 1) {
    factors.push(number);
  }
  return factors;
};

File: src/dataset/c/prime-factors_exercism-yorange1/gemini/GeminiFlash_prime_factors_v1.c
#include "prime_factors.h"
#include <math.h>

size_t find_factors(uint64_t n, uint64_t factors[static MAXFACTORS]) {
    size_t index = 0;

    // Handle 2 as a special case to optimize even number factorization
    while (n % 2 == 0) {
        factors[index++] = 2;
        n /= 2;
    }

    // Iterate only through odd numbers starting from 3
    for (uint64_t candidate = 3; candidate <= sqrt(n); candidate += 2) {
        while (n % candidate == 0) {
            factors[index++] = candidate;
            n /= candidate;
        }
    }

    // If n is still greater than 1, it means it's a prime number itself
    if (n > 1) {
        factors[index++] = n;
    }

    return index;
}